### PR TITLE
Integrate hosted emulator

### LINKED ISSUE #109: Integrate hosted emulator
## Instructions

Replace local emulator (terminal + local account data) with emulator hosted on the license server.
See https://github.com/onflow/cadence/pull/1739

Up to date architecture:
Screen Shot 2022-07-20 at 2 41 39 PM

### PR DESCRIPTION
**Pull request by DylanTinianov**
Closes #109 

## Description

Refactored the emulator controller to move the flow emulator to the language server.
These changes include:
- Implemented updated Language Server API: https://github.com/onflow/cadence/pull/1752
- The terminal to run the emulator locally is no longer needed
- Replaced local account data with a synchronized data response from the language server
  - Account Manager is no longer needed to interface with any local account data
- Removed start/ stop emulator commands
- Updated UI to reflect connected/ disconnected emulator status 

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: extension/src/utils/codelens.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {
+  Position,
+  Range,
+  window
+} from 'vscode'
+
+// This method will add and then remove a space on the last line to trick codelens to be updated
+export const refreshCodeLenses = (): void => {
```
Comment (sideninja) [inline extension/src/utils/codelens.ts:8]: Is there a better way? Could we just call the codeless API manually? Calling https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeLens

**Comment by sideninja** [comment]
Some UI feedback:
- on starting up the server there are these addresses output I find that unnecessary.
Screenshot 2022-08-16 at 12 36 15
- when switching account there's an output of the whole struct that's also probably not needed
Screenshot 2022-08-16 at 12 36 42
- also on switching account there's again an array of all addresses that can be removed 
Screenshot 2022-08-16 at 12 37 36
- if I switch accounts and click on the deploy contract codelens it gets deployed to the previous account, it looks like I have to click somewhere else in the editor first for that to get updated. Could you refresh as soon as account is switched.

I see there's problems with script execution, will look into LS side as well.

File: extension/src/utils/codelens.ts (hunk 1 above)
Comment (DylanTinianov) [reply extension/src/utils/codelens.ts:9]: Just fixed this so it works now. Previously it was always returning before doing the refresh!

**Comment by sideninja** [comment]
> > Some UI feedback:
> > 
> > * on starting up the server there are these addresses output I find that unnecessary.
> > 
> > Screenshot 2022-08-16 at 12 36 15
> > 
> > * when switching account there's an output of the whole struct that's also probably not needed
> > 
> > Screenshot 2022-08-16 at 12 36 42
> > 
> > * also on switching account there's again an array of all addresses that can be removed
> > 
> > Screenshot 2022-08-16 at 12 37 36
> > 
> > * if I switch accounts and click on the deploy contract codelens it gets deployed to the previous account, it looks like I have to click somewhere else in the editor first for that to get updated. Could you refresh as soon as account is switched.
> >   I see there's problems with script execution, will look into LS side as well.
> 
> Yeah the problem is that the extension isn't creating these pop up windows. I think VS Code is creating them automatically from the response of the LS.

I see. Will look into this on the LS side as well.

**Review by sideninja — APPROVED** [review]

**Pull request merged**